examples/cpp/hello_world/README.md

```diff
@@ -0,0 +1,184 @@
+# Hello world example
+
+The *eProsima Fast DDS hello world* example is a simple application intended to demonstrate a basic DDS deployment.
+
+## Introduction
+
+This example is part of the suite of examples designed by eProsima that aims to illustrate the features and possible configurations of DDS deployments through *eProsima Fast DDS*.
+
+In this case, the *hello world* example describes the simplest deployment of a Fast DDS publisher and subscriber.
+
+## Example in deep
+
+Each entity of the example (publisher and subscriber) creates different nested DDS entities: domain participant, publisher & dataWriter, and subscriber & dataReader, respectively.
+In both cases, the three DDS entities (domain participant, publisher/subscriber and dataWriter/dataReader) load their default configuration from the environment.
+If the environment does not specify the expected configuration, they take the default configuration per entity.
+For further information regarding the configuration environment, please refer to the *[XML profile playground](#xml-profile-playground)* subsection in the *[advanced configuration](#advanced-configuration)* section.
+
+The *hello world* example, together with the remain examples, would include a listening callback on the subscriber side. The subscriber will manage the new available data in the same thread as the main subscriber application.
+For simplicity, the ``HelloWorldSubscriber`` implements it own callback.
+
+Furthermore, this particular example includes a wait-set implementation. In contrast to implementing a listening callback, the wait-set is a mechanism where a dedicated thread waits until a status condition occurs. In that moment, that status condition triggering event would be evaluated to determine witch actions should be taken against it.
```

Comment:
This needs to be rephrased. I would not refer to the rest of the examples here, just to this one. I'd say that it can be run with two different subscription paradigms, and then explain the two